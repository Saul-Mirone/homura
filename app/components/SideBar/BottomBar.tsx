import React from 'react';
import { PlusIcon, RefreshIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export enum Step {
  EnterUrl,
  EnterName,
}

export type BottomBarProps = {
  step: Step | null;
  onClickPlus: () => void;
  onClickSync: () => void;
  render: (step: Step) => JSX.Element;
  refreshing: boolean;
};

export const BottomBar: React.FC<BottomBarProps> = ({
  step,
  onClickPlus,
  onClickSync,
  render,
  refreshing,
}) => (
  <div className="relative flex justify-between text-gray-300">
    {step !== null && (
      <div className="absolute bottom-0 inset-x-0 z-10">{render(step)}</div>
    )}
    <IconContainer onClick={onClickPlus}>
      <PlusIcon />
    </IconContainer>
    <IconContainer onClick={onClickSync}>
      <RefreshIcon spin={refreshing} />
    </IconContainer>
  </div>
);
