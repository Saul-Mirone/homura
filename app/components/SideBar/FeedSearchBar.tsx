import React from 'react';
import { SearchCircleFilledIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export type FeedSearchBarProps = {
  link: string;
  loading: boolean;
  onLinkChange: (link: string) => void;
  onSearch: (link: string) => void;
  onCancel: () => void;
};

export const FeedSearchBar: React.FC<FeedSearchBarProps> = ({
  onSearch,
  link,
  onLinkChange,
  onCancel,
  loading,
}) => {
  const inputEl = React.useRef<HTMLInputElement>(null);
  React.useEffect(() => {
    inputEl.current?.focus();
  }, []);

  return (
    <div className="flex bg-gray-300 text-gray-800">
      <div className="flex-1">
        <input
          ref={inputEl}
          value={link}
          onChange={(e) => onLinkChange(e.target.value)}
          onBlur={() => {
            onCancel();
          }}
          className="w-full p-2 text-xs bg-transparent"
          placeholder="Feed URL"
        />
      </div>
      <IconContainer
        className={`transition duration-300 ${
          !loading
            ? 'hover:bg-gray-700 hover:text-gray-300 text-gray-700'
            : 'bg-gray-600 text-gray-300'
        }`}
        disabled={loading}
        onClick={() => {
          if (loading) return;
          onSearch(link);
        }}
      >
        <SearchCircleFilledIcon />
      </IconContainer>
    </div>
  );
};
