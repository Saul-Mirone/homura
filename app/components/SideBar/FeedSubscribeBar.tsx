import React from 'react';
import { CheckCircleFilledIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export type FeedSubscribeBarProps = {
  link: string;
  name: string;
  onNameChange: (name: string) => void;
  onConfirm: (name: string) => void;
  onCancel: () => void;
};

export const FeedSubscribeBar: React.FC<FeedSubscribeBarProps> = ({
  link,
  name,
  onNameChange,
  onConfirm,
  onCancel,
}) => {
  const inputEl = React.useRef<HTMLInputElement>(null);
  React.useEffect(() => {
    inputEl.current?.focus();
  }, []);

  return (
    <>
      <div className="p-2 text-xs truncate">{link}</div>
      <div className="flex bg-gray-300 text-gray-800">
        <div className="flex-1">
          <input
            ref={inputEl}
            value={name}
            onChange={(e) => onNameChange(e.target.value)}
            onBlur={() => {
              onCancel();
            }}
            className="w-full p-2 text-xs bg-transparent"
            placeholder="Save as Name"
          />
        </div>
        <IconContainer
          onClick={() => onConfirm(name)}
          className="transition duration-300 text-gray-700 hover:bg-gray-700 hover:text-gray-300"
        >
          <CheckCircleFilledIcon />
        </IconContainer>
      </div>
    </>
  );
};
