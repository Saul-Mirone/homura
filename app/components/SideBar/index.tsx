import React from 'react';

export type SideBarProps = {
  overview: JSX.Element;
  bottom: JSX.Element;
  onClick: () => void;
};

export const SideBar: React.FC<SideBarProps> = ({
  overview,
  bottom,
  children,
  onClick,
}) => {
  return (
    <div
      role="presentation"
      onClick={onClick}
      className="w-1/6 bg-gray-700 pt-10 flex flex-col h-screen select-none"
    >
      <div className="flex-1 thin-scroll">
        {overview}
        <div className="pt-10" />
        {children}
      </div>
      {bottom}
    </div>
  );
};
