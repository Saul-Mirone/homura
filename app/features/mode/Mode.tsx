import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Toolkit } from '../../components/SubSideBar/ToolKit';
import { AppDispatch } from '../../store';
import { selectMode, switchMode } from './modeSlice';

export const Mode: React.FC = () => {
  const dispatch = useDispatch<AppDispatch>();
  const mode = useSelector(selectMode);

  return <Toolkit mode={mode} onSwitchMode={(m) => dispatch(switchMode(m))} />;
};
