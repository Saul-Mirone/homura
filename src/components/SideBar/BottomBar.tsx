import React from 'react';
import { PlusIcon, RefreshIcon } from '../Icon';
import { IconContainer } from '../LogoIcon';

export enum Step {
  EnterUrl,
  EnterName,
}

export type BottomBarProps = {
  step: Step | undefined;
  onClickPlus: () => void;
  onClickSync: () => void;
  render: (step: Step) => JSX.Element;
  loading: boolean;
};

export const BottomBar: React.FC<BottomBarProps> = ({
  step,
  onClickPlus,
  onClickSync,
  render,
  loading,
}) => (
  <div role="toolbar" className="sidebar-bottom-bar">
    {step !== undefined && (
      <div className="sidebar-bottom-bar__step">{render(step)}</div>
    )}
    <IconContainer onClick={onClickPlus}>
      <PlusIcon />
    </IconContainer>
    <IconContainer onClick={onClickSync}>
      <RefreshIcon spin={loading} />
    </IconContainer>
  </div>
);
