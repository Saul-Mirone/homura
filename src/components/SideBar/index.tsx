import React from 'react';
import './style.pcss';

export type SideBarProps = {
    overview: JSX.Element;
    bottom: JSX.Element;
    onClick: () => void;
};

export const SideBar: React.FC<SideBarProps> = ({ overview, bottom, children, onClick }) => (
    <div
        role="menu"
        tabIndex={0}
        aria-label="Sources"
        onClick={onClick}
        onKeyDown={onClick}
        className="sidebar-container"
    >
        <div className="relative flex-1 thin-scroll">
            {overview}
            <div className="pt-10" />
            {children}
        </div>
        {bottom}
    </div>
);
