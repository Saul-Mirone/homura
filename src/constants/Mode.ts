export enum Mode {
    All = 'All',
    Unread = 'Unread',
    Starred = 'Starred',
}
