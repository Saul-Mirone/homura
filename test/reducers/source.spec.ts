import { AnyAction } from 'redux';

// organize-imports-ignore
import { mockChannel } from '../test-tools/mockChannel';

import { Step } from '../../src/components/SideBar/BottomBar';
import { Mode } from '../../src/constants/Mode';
import { Status } from '../../src/constants/Status';
import {
  fetchSources,
  resetSubscribeError,
  resetSubscribeState,
  searchUrlForSource,
  setCurrentSource,
  showSubscribeBar,
  sourceReducer,
  State,
  subscribeToSource,
  syncSources,
  unsubscribeById,
  updateSourceById,
} from '../../src/features/source/sourceSlice';
import { formatActions } from '../test-tools/formatActions';
import { mockStore } from '../test-tools/mockStore';

describe('reducers', () => {
  describe('source', () => {
    it('should handle initial state', () => {
      expect(sourceReducer(undefined, {} as AnyAction)).toMatchSnapshot();
    });

    it('should handle setCurrentSource', () => {
      expect(
        sourceReducer({ activeId: null } as State, {
          type: setCurrentSource,
          payload: 1,
        })
      ).toMatchSnapshot();
    });

    it('should handle showSubscribeBar', () => {
      expect(
        sourceReducer(
          { activeId: null, subscribeStatus: Status.Succeeded } as State,
          {
            type: showSubscribeBar,
          }
        )
      ).toMatchSnapshot();
    });

    it('should handle resetSubscribeError', () => {
      expect(
        sourceReducer({ subscribeError: 'subscribeError' } as State, {
          type: resetSubscribeError,
        })
      ).toMatchSnapshot();
    });

    it('should handle resetSubscribeState', () => {
      expect(
        sourceReducer(
          {
            subscribeStep: Step.EnterName,
            subscribeLink: 'fake-link',
            subscribeName: 'fake-name',
            subscribeError: 'subscribeError',
            subscribeStatus: Status.Succeeded,
          } as State,
          {
            type: resetSubscribeState,
          }
        )
      ).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(
        sourceReducer({ subscribeStep: null } as State, { type: 'unknown' })
      ).toMatchSnapshot();
    });

    describe('thunk', () => {
      describe('fetchSources', () => {
        it('should action list match snapshot', async () => {
          const store = mockStore();
          mockChannel.getSourceList.mockReturnValue([
            {
              id: 1,
              name: 'data-1',
              link: 'link-1',
              count: 2,
            },
            {
              id: 2,
              icon: 'test-icon',
              name: 'data-2',
              link: 'link-2',
              count: 10,
            },
          ]);
          await store.dispatch(fetchSources(Mode.All));
          expect(formatActions(store)).toMatchSnapshot();
        });

        it('should handle fetchSources.pending', () => {
          expect(
            sourceReducer(
              {
                fetchListStatus: Status.Idle,
              } as State,
              {
                type: fetchSources.pending,
              }
            )
          ).toMatchSnapshot();
        });

        it('should handle fetchSources.fulfilled', () => {
          expect(
            sourceReducer(
              {
                fetchListStatus: Status.Idle,
              } as State,
              {
                type: fetchSources.fulfilled,
                payload: [
                  {
                    name: 'fake-data',
                  },
                ],
              }
            )
          ).toMatchSnapshot();
        });

        it('should handle fetchSources.rejected', () => {
          expect(
            sourceReducer(
              {
                fetchListStatus: Status.Idle,
              } as State,
              {
                type: fetchSources.rejected,
              }
            )
          ).toMatchSnapshot();
        });
      });

      describe('syncSources', () => {
        it('should action list match snapshot', async () => {
          const store = mockStore();
          mockChannel.getSourceList.mockReturnValue([
            {
              id: 1,
              name: 'data-1-sync',
              link: 'link-1-sync',
              count: 2,
            },
            {
              id: 2,
              icon: 'test-icon',
              name: 'data-2-sync',
              link: 'link-2-sync',
              count: 10,
            },
          ]);
          await store.dispatch(syncSources(Mode.All));
          expect(formatActions(store)).toMatchSnapshot();
        });

        it('should handle syncSources.pending', () => {
          expect(
            sourceReducer(
              {
                fetchListStatus: Status.Idle,
              } as State,
              {
                type: syncSources.pending,
              }
            )
          ).toMatchSnapshot();
        });

        it('should handle syncSources.fulfilled', () => {
          expect(
            sourceReducer(
              {
                fetchListStatus: Status.Idle,
              } as State,
              {
                type: syncSources.fulfilled,
                payload: [
                  {
                    name: 'fake-data',
                  },
                ],
              }
            )
          ).toMatchSnapshot();
        });

        it('should handle syncSources.rejected', () => {
          expect(
            sourceReducer(
              {
                fetchListStatus: Status.Idle,
              } as State,
              {
                type: syncSources.rejected,
              }
            )
          ).toMatchSnapshot();
        });
      });

      describe('unsubscribeById', () => {
        it('should action list match snapshot', async () => {
          const store = mockStore();
          await store.dispatch(unsubscribeById(1));
          expect(formatActions(store)).toMatchSnapshot();
        });

        it('should handle unsubscribeById.fulfilled', () => {
          expect(
            sourceReducer(
              {
                activeId: 2,
                list: [
                  { id: 1, name: 'data-1' },
                  { id: 2, name: 'data-2' },
                ],
              } as State,
              {
                type: unsubscribeById.fulfilled,
                payload: 2,
              }
            )
          ).toMatchSnapshot();
        });
      });

      describe('updateSourceById', () => {
        it('should action list match snapshot', async () => {
          const store = mockStore();
          await store.dispatch(updateSourceById({ id: 1, name: 'next-name' }));
          expect(formatActions(store)).toMatchSnapshot();
        });

        it('should handle updateSourceById.fulfilled', () => {
          expect(
            sourceReducer(
              {
                activeId: 2,
                list: [
                  { id: 1, name: 'data-1' },
                  { id: 2, name: 'data-2' },
                ],
              } as State,
              {
                type: updateSourceById.fulfilled,
                payload: {
                  id: 2,
                  name: 'next-data-2',
                },
              }
            )
          ).toMatchSnapshot();
        });
      });

      describe('searchUrlForSource', () => {
        it('should action list match snapshot', async () => {
          const store = mockStore();
          await store.dispatch(searchUrlForSource('fake-link'));
          expect(formatActions(store)).toMatchSnapshot();
        });

        it('should success action list match snapshot', async () => {
          const store = mockStore();
          mockChannel.checkUrl.mockResolvedValueOnce('fake-result');
          await store.dispatch(searchUrlForSource('fake-link'));
          expect(formatActions(store)).toMatchSnapshot();
        });

        it('should rejected action list match snapshot', async () => {
          const store = mockStore();
          mockChannel.checkUrl.mockResolvedValue('');
          await store.dispatch(searchUrlForSource('fake-link'));
          expect(formatActions(store)).toMatchSnapshot();
        });

        it('should handle searchUrlForSource.pending', () => {
          expect(
            sourceReducer(
              {
                subscribeLink: '',
                subscribeStatus: Status.Idle,
              } as State,
              {
                type: searchUrlForSource.pending,
                meta: {
                  arg: 'fake-link',
                },
              }
            )
          ).toMatchSnapshot();
        });

        it('should handle searchUrlForSource.fulfilled', () => {
          expect(
            sourceReducer(
              {
                subscribeName: '',
                subscribeStatus: Status.Idle,
              } as State,
              {
                type: searchUrlForSource.fulfilled,
                payload: 'fake-name',
              }
            )
          ).toMatchSnapshot();
        });

        it('should handle searchUrlForSource.rejected', () => {
          expect(
            sourceReducer(
              {
                subscribeError: null,
                subscribeStatus: Status.Idle,
              } as State,
              {
                type: searchUrlForSource.rejected,
                error: {
                  message: 'fake-error',
                },
              }
            )
          ).toMatchSnapshot();
        });

        it('should handle searchUrlForSource.rejected even when no message', () => {
          expect(
            sourceReducer(
              {
                subscribeError: null,
                subscribeStatus: Status.Idle,
              } as State,
              {
                type: searchUrlForSource.rejected,
                error: {},
              }
            )
          ).toMatchSnapshot();
        });
      });

      describe('subscribeToSource', () => {
        it('should action list match snapshot', async () => {
          const store = mockStore();
          mockChannel.confirm.mockReturnValue({
            id: 4,
            name: '',
            link: 'fake-link',
            count: 5,
          });
          await store.dispatch(
            subscribeToSource({ name: 'fake-name', mode: Mode.All })
          );
          expect(formatActions(store)).toMatchSnapshot();
        });

        it('should action list match snapshot when mode is Starred', async () => {
          const store = mockStore();
          mockChannel.confirm.mockReturnValue({
            id: 4,
            name: '',
            link: 'fake-link',
            count: 5,
          });
          await store.dispatch(
            subscribeToSource({ name: 'fake-name', mode: Mode.Starred })
          );
          expect(formatActions(store)).toMatchSnapshot();
        });

        it('should handle subscribeToSource.fulfilled', () => {
          expect(
            sourceReducer(
              {
                activeId: 4,
                subscribeError: 'fake-error',
                subscribeStatus: Status.Idle,
                subscribeLink: 'fake-link',
                subscribeName: 'fake-name',
                list: [{ id: 4 }],
              } as State,
              {
                type: subscribeToSource.fulfilled,
                payload: {
                  id: 1,
                  name: 'fake-data-1',
                },
              }
            )
          ).toMatchSnapshot();
        });
      });
    });
  });
});
