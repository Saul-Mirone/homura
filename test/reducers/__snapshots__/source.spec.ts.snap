// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers source should handle initial state 1`] = `
Object {
  "activeId": undefined,
  "fetchListStatus": 0,
  "list": Array [],
  "subscribeError": undefined,
  "subscribeLink": "",
  "subscribeName": "",
  "subscribeStatus": 0,
  "subscribeStep": undefined,
  "syncListStatus": 0,
}
`;

exports[`reducers source should handle resetSubscribeError 1`] = `
Object {
  "subscribeError": undefined,
}
`;

exports[`reducers source should handle resetSubscribeState 1`] = `
Object {
  "subscribeError": undefined,
  "subscribeLink": "",
  "subscribeName": "",
  "subscribeStatus": 0,
  "subscribeStep": undefined,
}
`;

exports[`reducers source should handle setCurrentSource 1`] = `
Object {
  "activeId": 1,
}
`;

exports[`reducers source should handle showSubscribeBar 1`] = `
Object {
  "activeId": null,
  "subscribeStatus": 0,
  "subscribeStep": 0,
}
`;

exports[`reducers source should handle unknown action type 1`] = `
Object {
  "subscribeStep": null,
}
`;

exports[`reducers source thunk fetchSources should action list match snapshot 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": 0,
      "requestId": "fake-request-id",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "source/fetchSources/pending",
  },
  Object {
    "meta": Object {
      "arg": 0,
      "requestId": "fake-request-id",
      "requestStatus": "fulfilled",
    },
    "payload": Array [
      Object {
        "count": 2,
        "id": 1,
        "link": "link-1",
        "name": "data-1",
      },
      Object {
        "count": 10,
        "icon": "test-icon",
        "id": 2,
        "link": "link-2",
        "name": "data-2",
      },
    ],
    "type": "source/fetchSources/fulfilled",
  },
]
`;

exports[`reducers source thunk fetchSources should handle fetchSources.fulfilled 1`] = `
Object {
  "fetchListStatus": 2,
  "list": Array [
    Object {
      "name": "fake-data",
    },
  ],
}
`;

exports[`reducers source thunk fetchSources should handle fetchSources.pending 1`] = `
Object {
  "fetchListStatus": 1,
}
`;

exports[`reducers source thunk fetchSources should handle fetchSources.rejected 1`] = `
Object {
  "fetchListStatus": 3,
}
`;

exports[`reducers source thunk searchUrlForSource should action list match snapshot 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": "fake-link",
      "requestId": "fake-request-id",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "source/searchUrlForSource/pending",
  },
  Object {
    "error": Object {
      "message": "ParseRSSFailed",
      "name": "Error",
      "stack": "fake-stack",
    },
    "meta": Object {
      "aborted": false,
      "arg": "fake-link",
      "condition": false,
      "rejectedWithValue": false,
      "requestId": "fake-request-id",
      "requestStatus": "rejected",
    },
    "payload": undefined,
    "type": "source/searchUrlForSource/rejected",
  },
]
`;

exports[`reducers source thunk searchUrlForSource should handle searchUrlForSource.fulfilled 1`] = `
Object {
  "subscribeName": "fake-name",
  "subscribeStatus": 0,
  "subscribeStep": 1,
}
`;

exports[`reducers source thunk searchUrlForSource should handle searchUrlForSource.pending 1`] = `
Object {
  "subscribeLink": "fake-link",
  "subscribeStatus": 1,
}
`;

exports[`reducers source thunk searchUrlForSource should handle searchUrlForSource.rejected 1`] = `
Object {
  "subscribeError": "fake-error",
  "subscribeStatus": 3,
}
`;

exports[`reducers source thunk searchUrlForSource should handle searchUrlForSource.rejected even when no message 1`] = `
Object {
  "subscribeError": "Unknown Error",
  "subscribeStatus": 3,
}
`;

exports[`reducers source thunk searchUrlForSource should rejected action list match snapshot 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": "fake-link",
      "requestId": "fake-request-id",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "source/searchUrlForSource/pending",
  },
  Object {
    "error": Object {
      "message": "ParseRSSFailed",
      "name": "Error",
      "stack": "fake-stack",
    },
    "meta": Object {
      "aborted": false,
      "arg": "fake-link",
      "condition": false,
      "rejectedWithValue": false,
      "requestId": "fake-request-id",
      "requestStatus": "rejected",
    },
    "payload": undefined,
    "type": "source/searchUrlForSource/rejected",
  },
]
`;

exports[`reducers source thunk searchUrlForSource should success action list match snapshot 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": "fake-link",
      "requestId": "fake-request-id",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "source/searchUrlForSource/pending",
  },
  Object {
    "meta": Object {
      "arg": "fake-link",
      "requestId": "fake-request-id",
      "requestStatus": "fulfilled",
    },
    "payload": "fake-result",
    "type": "source/searchUrlForSource/fulfilled",
  },
]
`;

exports[`reducers source thunk subscribeToSource should action list match snapshot 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": Object {
        "mode": 0,
        "name": "fake-name",
      },
      "requestId": "fake-request-id",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "source/subscribeToSource/pending",
  },
  Object {
    "meta": Object {
      "arg": Object {
        "mode": 0,
        "name": "fake-name",
      },
      "requestId": "fake-request-id",
      "requestStatus": "fulfilled",
    },
    "payload": Object {
      "count": 5,
      "icon": undefined,
      "id": 4,
      "link": "fake-link",
      "name": "fake-name",
    },
    "type": "source/subscribeToSource/fulfilled",
  },
]
`;

exports[`reducers source thunk subscribeToSource should action list match snapshot when mode is Starred 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": Object {
        "mode": 2,
        "name": "fake-name",
      },
      "requestId": "fake-request-id",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "source/subscribeToSource/pending",
  },
  Object {
    "meta": Object {
      "arg": Object {
        "mode": 2,
        "name": "fake-name",
      },
      "requestId": "fake-request-id",
      "requestStatus": "fulfilled",
    },
    "payload": Object {
      "count": 0,
      "icon": undefined,
      "id": 4,
      "link": "fake-link",
      "name": "fake-name",
    },
    "type": "source/subscribeToSource/fulfilled",
  },
]
`;

exports[`reducers source thunk subscribeToSource should handle subscribeToSource.fulfilled 1`] = `
Object {
  "activeId": 1,
  "list": Array [
    Object {
      "id": 4,
    },
    Object {
      "id": 1,
      "name": "fake-data-1",
    },
  ],
  "subscribeError": undefined,
  "subscribeLink": "",
  "subscribeName": "",
  "subscribeStatus": 2,
  "subscribeStep": undefined,
}
`;

exports[`reducers source thunk syncSources should action list match snapshot 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": 0,
      "requestId": "fake-request-id",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "source/syncSources/pending",
  },
  Object {
    "meta": Object {
      "arg": 0,
      "requestId": "fake-request-id",
      "requestStatus": "fulfilled",
    },
    "payload": undefined,
    "type": "source/syncSources/fulfilled",
  },
]
`;

exports[`reducers source thunk syncSources should handle syncSources.fulfilled 1`] = `
Object {
  "fetchListStatus": 0,
  "syncListStatus": 2,
}
`;

exports[`reducers source thunk syncSources should handle syncSources.pending 1`] = `
Object {
  "fetchListStatus": 0,
  "syncListStatus": 1,
}
`;

exports[`reducers source thunk syncSources should handle syncSources.rejected 1`] = `
Object {
  "fetchListStatus": 0,
  "syncListStatus": 3,
}
`;

exports[`reducers source thunk unsubscribeById should action list match snapshot 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": 1,
      "requestId": "fake-request-id",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "source/unsubscribeById/pending",
  },
  Object {
    "meta": Object {
      "arg": 1,
      "requestId": "fake-request-id",
      "requestStatus": "fulfilled",
    },
    "payload": 1,
    "type": "source/unsubscribeById/fulfilled",
  },
]
`;

exports[`reducers source thunk unsubscribeById should handle unsubscribeById.fulfilled 1`] = `
Object {
  "activeId": undefined,
  "list": Array [
    Object {
      "id": 1,
      "name": "data-1",
    },
  ],
}
`;

exports[`reducers source thunk updateSourceById should action list match snapshot 1`] = `
Array [
  Object {
    "meta": Object {
      "arg": Object {
        "id": 1,
        "name": "next-name",
      },
      "requestId": "fake-request-id",
      "requestStatus": "pending",
    },
    "payload": undefined,
    "type": "source/updateSourceById/pending",
  },
  Object {
    "meta": Object {
      "arg": Object {
        "id": 1,
        "name": "next-name",
      },
      "requestId": "fake-request-id",
      "requestStatus": "fulfilled",
    },
    "payload": Object {
      "id": 1,
      "name": "next-name",
    },
    "type": "source/updateSourceById/fulfilled",
  },
]
`;

exports[`reducers source thunk updateSourceById should handle updateSourceById.fulfilled 1`] = `
Object {
  "activeId": undefined,
  "list": Array [
    Object {
      "id": 1,
      "name": "data-1",
    },
    Object {
      "id": 2,
      "name": "next-data-2",
    },
  ],
}
`;
